owncloudApp/src/main/java/com/owncloud/android/presentation/UIResult.kt

```diff
@@ -41,6 +43,15 @@ sealed class UIResult<out T> {
             is Error -> error
             else -> null
         }
+
+    companion object {
+        fun <T> fromResult(result: UseCaseResult<T>): UIResult<T> =
+            if (result.isSuccess) {
+                Success(result.getDataOrNull())
```

Comment:
If we use `result is Success` instead of `result.isSuccess`, the cast inside will be smart 😉 

```suggestion
            if (result is Success) {
                Success(result.data)
```

Reply:
Same here.